src/Grid/topology.jl

```diff
@@ -0,0 +1,341 @@
+    cell_edges = edges(getcells(grid,cellid))
+    nonlocal_edgeid = cell_edges[local_edgeidx]
+    cell_neighbors = getneighborhood(top,grid,CellIndex(cellid))
+    self_reference_local = EdgeIndex[]
+    for cellid in cell_neighbors
+        local_neighbor_edgeid = findfirst(x->issubset(x,nonlocal_edgeid),edges(getcells(grid,cellid)))::Int
+        local_neighbor_edgeid === nothing && continue
+        local_edge = EdgeIndex(cellid,local_neighbor_edgeid)
+        push!(self_reference_local, local_edge)
+    end
+    if include_self
+        return unique([top.edge_edge_neighbor[cellid, local_edgeidx].neighbor_info; self_reference_local; edgeidx])
+    else
+        return unique([top.edge_edge_neighbor[cellid, local_edgeidx].neighbor_info; self_reference_local])
+    end
+end
+
+"""
+    compute_vertex_star_stencils(top, grid) -> StencilType
+Computes the stencils induced by the edge connectivity of the vertices.
+"""
+function compute_vertex_star_stencils(top::ExclusiveTopology, grid::Grid)
+    cells = grid.cells
+    stencil_table = Dict{Int,EntityNeighborhood{VertexIndex}}()
+    # Vertex Connectivity
+    for (global_vertexid,cellset) ∈ enumerate(top.vertex_to_cell)
+        vertex_neighbors_local = VertexIndex[]
+        for cell ∈ cellset
+            neighbor_boundary = getdim(cells[cell]) > 2 ? collect(edges(cells[cell])) : collect(faces(cells[cell])) #get lowest dimension boundary
+            neighbor_connected_faces = neighbor_boundary[findall(x->global_vertexid ∈ x, neighbor_boundary)]
+            this_local_vertex = findfirst(i->toglobal(grid, VertexIndex(cell, i)) == global_vertexid, 1:nvertices(cells[cell]))
+            push!(vertex_neighbors_local, VertexIndex(cell, this_local_vertex))
+            other_vertices = findfirst.(x->x!=global_vertexid,neighbor_connected_faces)
+            any(other_vertices .=== nothing) && continue
+            neighbor_vertices_global = getindex.(neighbor_connected_faces, other_vertices)
+            neighbor_vertices_local = [VertexIndex(cell,local_vertex) for local_vertex ∈ findall(x->x ∈ neighbor_vertices_global, vertices(cells[cell]))]
+            append!(vertex_neighbors_local, neighbor_vertices_local)
+        end
+        stencil_table[global_vertexid] =  EntityNeighborhood(vertex_neighbors_local)
+    end
+    return stencil_table
+end
+
+"""
+    getstencil(top::StencilType, grid::AbstractGrid, vertex_idx::VertexIndex) -> Iterable{EntityNeighborhood}
+Get an iterateable over the stencil members for a given local entity.
+"""
+function getstencil(top::Dict{Int,EntityNeighborhood{VertexIndex}}, grid::AbstractGrid, vertex_idx::VertexIndex)
+    return top[toglobal(grid, vertex_idx)].neighbor_info
+end
+
+"""
+    compute_face_skeleton(topology, grid) -> Iterable{FaceIndex}
+Creates an iterateable face skeleton. The skeleton consists of `FaceIndex` that can be used to `reinit`
+`FaceValues`.
+"""
+function compute_face_skeleton(top::ExclusiveTopology, grid::Grid)
```

Comment:
why is this outside of the struct?

Reply:
Two reasons here. First, the skeleton is not always needed if we query basic topological information and second the algorithm is actually separate. I think this is also what @fredrikekre had in mind here.

Reply:
How about a keyword argument in the constructor and have a union with nothing in the struct?